src/transforms/functional.jl

```diff
@@ -28,18 +39,60 @@ inverse(::typeof(cos))  = acos
 inverse(::typeof(acos)) = cos
 inverse(::typeof(sin))  = asin
 inverse(::typeof(asin)) = sin
+inverse(::typeof(cosd))  = acosd
+inverse(::typeof(acosd)) = cosd
+inverse(::typeof(sind))  = asind
+inverse(::typeof(asind)) = sind
+inverse(::typeof(identity)) = identity
 
 # fallback to nothing
 inverse(::Any) = nothing
 
-colcache(::Functional, x) = nothing
+isrevertible(transform::Functional) =
+  !isnothing(inverse(transform.func))
+
+isrevertible(transform::Functional{<:NamedTuple}) =
+  all(!isnothing, inverse.(values(transform.func)))
 
-function colapply(transform::Functional, x, c)
-  f = transform.func
-  f.(x)
+function applyfunc(transform::Functional, cols, nm)
+  x = Tables.getcolumn(cols, nm)
+  func = transform.func
+  func.(x)
 end
 
-function colrevert(transform::Functional, y, c)
-  g = inverse(transform.func)
-  g.(y)
-end
\ No newline at end of file
+function applyfunc(transform::Functional{<:NamedTuple}, cols, nm)
```

Comment:
I find this design with extra functions overly complicated. Can we refactor the code so that we have a single apply and revert? Maybe a single helper function?

```julia
_functuple(func, table) = (; Tables.columnanes(table) .=> func)
_functuple(func::NamedTuple, table) = func
```

All the rest could assume a named tuple always?